sklearn/decomposition/_sparse_pca.py

```diff
@@ -29,6 +29,51 @@ def _check_normalize_components(normalize_components, estimator_name):
             )
 
 
+def _get_explained_variance(X, components, ridge_alpha):
+    """Get the explained variance.
+
+    Get the explained variance from the principal components of the
+    data. This follows the method outlined in [1] section 3.4 (Adjusted Total
+    Variance). For an alternate approach (not implemented here), see [2].
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_samples, n_features)
+        The feature vector. n_samples and n_features are the number of
+        samples and features, respectively.
+
+    components : array, shape (n_components, n_features)
+        The principal components. [1]
+
+    Notes
+    -----
+    Orthogonality is enforced in this case. Other variants exist that don't
+    enforce this [2].
+
+    References
+    ----------
+    .. [1] Hui Zou, Trevor Hastie, Robert Tibshirani ; Journal of Computational
+        and Graphical Statistics, Volume 15, Number 2, Pages 265–286.
+        DOI: 10.1198/106186006X113430
+    .. [2] Rodolphe Jenatton, Guillaume Obozinski, Francis Bach ; Proceedings
+        of the Thirteenth International Conference on Artificial Intelligence
+        and Statistics (AISTATS), PMLR 9:366-373, 2010.
```

Comment:
please put titles of papers. Some examples:

https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/_forest.py#L1104
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/_bagging.py#L565